src/irr/asset/CImageWriterOpenEXR.cpp

```diff
@@ -90,12 +90,12 @@ namespace irr
 				for (uint64_t yPos = region->imageOffset.y; yPos < region->imageOffset.y + regionHeight; ++yPos)
 					for (uint64_t xPos = region->imageOffset.x; xPos < region->imageOffset.x + regionWidth; ++xPos)
 					{
-						const uint64_t ptrStyleEndShiftToImageDataPixel = ((regionHeight + yPos) * regionWidth + xPos) * blockByteSize + region->bufferOffset;
+						const uint8_t* texelPtr = reinterpret_cast<const uint8_t*>(data) + region->bufferOffset + (yPos * regionWidth + xPos) * asset::getTexelOrBlockBytesize(image->getCreationParameters().format);
```

Comment:
better call `getTexelOrBlockBytesize` once and save to some var. This call is quite big switch so maybe better no to call it in a loop

Reply:
you're right, I saved it